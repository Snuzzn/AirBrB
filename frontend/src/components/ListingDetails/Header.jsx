import React from 'react'
import { HiStar } from 'react-icons/hi'
import { Link } from 'react-scroll'
import PropTypes from 'prop-types';

function Header ({ listingInfo }) {
  return (
    <div className="flex items-center  text-gray-700 justify-between">
      <div>
        <p className="text-3xl font-medium">{listingInfo.title}</p>
        <p className="text-gray-500">{listingInfo.address.street}
          <span className="text-gray-500"> Â· {listingInfo.address.city}</span>
        </p>
      </div>
      <div className="flex flex-col items-end">
        <div className="flex items-center gap-2 ">
          <HiStar className="text-red-400 text-xl"/>
          <p className="text-gray-600 font-medium text-2xl">4.96</p>
        </div>
        <Link to="reviews" smooth={true} duration={1000}>
          <p className="text-sm text-gray-500 hover:text-red-400 cursor-pointer">12 reviews</p>
        </Link>
      </div>
    </div>
  )
}

export default Header

Header.propTypes = {
  listingInfo: PropTypes.object,
}
