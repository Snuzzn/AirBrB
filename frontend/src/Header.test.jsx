import React from 'react';
import Header from './components/ListingDetails/Header';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Header', () => {
  /**
   * Header takes in a listing info object, and then
   * extracts and displays data for the listing title,
   * address and reviews,
   */

  // Note that for all fields except 'reviews', the website
  // will not allow the user to enter empty strings
  const listingInfo = {
    title: 'Simpsons Residence',
    address: {
      street: '742 Evergreen Terrace',
      city: 'Springfield'
    },
    reviews: [
      {
        score: 3,
        review: 'Sofa was really old'
      },
      {
        score: 4,
        review: 'That neighbour Ned is a bit odd'
      }
    ]
  }

  it('renders with listing title', () => {
    const header = shallow(<Header listingInfo={listingInfo} />);
    expect(header.find('#header-title').text()).toBe(listingInfo.title);
  })

  it('renders with listing street', () => {
    const header = shallow(<Header listingInfo={listingInfo} />);
    expect(header.find('#header-street').text()).toContain(listingInfo.address.street);
  })

  it('renders with listing city', () => {
    const header = shallow(<Header listingInfo={listingInfo} />);
    expect(header.find('#header-city').text()).toContain(listingInfo.address.city);
  })

  it('renders with number of reviews', () => {
    const header = shallow(<Header listingInfo={listingInfo} />);
    expect(header.find('#header-reviews').text()).toBe('2 reviews');
  })

  it('calculates average score and rounds to 2 decimal place', () => {
    const header = shallow(<Header listingInfo={listingInfo} />);
    expect(header.find('#header-tooltip').prop('score')).toBe('3.5');
  })

  it('uses hyphen if no review scores exist', () => {
    listingInfo.reviews = [];
    const header = shallow(<Header listingInfo={listingInfo} />);
    expect(header.find('#header-tooltip').prop('score')).toBe('-');
  })
})
