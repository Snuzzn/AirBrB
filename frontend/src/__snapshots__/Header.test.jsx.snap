// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders with all listing info data 1`] = `
<div
  className="flex items-center  text-gray-700 justify-between"
>
  <div>
    <p
      className="flex items-center text-sm text-gray-500"
    >
       
      <button
        className="hover:underline"
      >
        Listings
      </button>
       
      <svg
        className="inline m-0"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M184 112l144 144-144 144"
          fill="none"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="48"
        />
      </svg>
      Details
    </p>
    <p
      className="text-3xl font-medium"
      id="header-title"
    >
      Simpsons Residence
    </p>
    <p
      className="text-gray-500 "
      id="header-street"
    >
      742 Evergreen Terrace
      <span
        className="text-gray-500"
        id="header-city"
      >
         Â· 
        Springfield
      </span>
    </p>
  </div>
  <div
    className="flex flex-col items-end"
  >
    <div
      className=""
      style={
        Object {
          "display": "inline",
        }
      }
    >
      <div
        className="flex items-center gap-2 cursor-pointer"
      >
        <svg
          className="text-red-400 text-xl"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
          />
        </svg>
        <p
          className="text-gray-600 font-medium text-2xl hover:text-red-400"
        >
          -
        </p>
      </div>
    </div>
    <a
      onClick={[Function]}
    >
      <p
        className="text-sm text-gray-500 hover:text-red-400 cursor-pointer"
        id="header-reviews"
      >
        0
         reviews
      </p>
    </a>
  </div>
</div>
`;
